sqlglot/generator.py

```diff
@@ -2345,6 +2346,18 @@ def ordered_sql(self, expression: exp.Ordered) -> str:
                     f"'{nulls_sort_change.strip()}' translation not supported in window functions"
                 )
                 nulls_sort_change = ""
+            elif (
+                self.NULL_ORDERING_SUPPORTED is False
+                and (isinstance(expression.find_ancestor(exp.AggFunc, exp.Select), exp.AggFunc))
+                and (
+                    (asc and nulls_sort_change == " NULLS LAST")
+                    or (desc and nulls_sort_change == " NULLS FIRST")
```

Comment:
What is this part checking for? What's the reasoning behind checking `asc` and `nulls_sort_change`?

Reply:
Aggregate functions actually allow `ASC NULLS FIRST` and `DESC NULLS LAST` (default behavior of `ASC` and `DESC` respectively) but not the other way around

Reply:
> which is the default behavior of `ASC` and `DESC` respectively

It looks like there may be a hidden assumption here. This isn't the case in general, which is why we have the `NULL_ORDERING` dialect constant. Did you take this into account? The base generator code should be generic, not just BQ-specific (if that was the rationale).

Reply:
Gotcha, yeah that makes sense. I was lacking the surrounding context in code. Sounds good, feel free to merge when ready 👍